src/shogun/machine/BaggingMachine.cpp

```diff
@@ -44,17 +47,59 @@ CBaggingMachine::~CBaggingMachine()
 
 CBinaryLabels* CBaggingMachine::apply_binary(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> output = apply_outputs_without_combination(data);
+
+	CMeanRule* mean_rule = new CMeanRule();
+
+	SGVector<float64_t> labels = m_combination_rule->combine(output);
+	SGVector<float64_t> probabilities = mean_rule->combine(output);
+
+	float64_t threshold = 0.5;
+	CBinaryLabels* pred = new CBinaryLabels(probabilities, threshold);
+
+	SG_UNREF(mean_rule);
 
-	CBinaryLabels* pred = new CBinaryLabels(combined_vector);
 	return pred;
 }
 
 CMulticlassLabels* CBaggingMachine::apply_multiclass(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> bagged_outputs =
+	    apply_outputs_without_combination(data);
+
+	auto labels_multiclass = dynamic_cast<CMulticlassLabels*>(m_labels);
+	REQUIRE(labels_multiclass, "Labels are not multiclass\n");
```

Comment:
minor: missing period at end of sentence.
Better error message
`REQUIRE(labels_multiclass, "Labels (%s) are not compatible with multiclass.\n", m_labels->get_name());`
(This needs a `REQUIRE(m_labels, "Labels not set.\n");` to avoid segfaults for NULL labels